src/docs/arch-metadata.dox

```diff
@@ -1,19 +1,78 @@
 /*! @arch_page arch-metadata Metadata
 
-Metadata in WiredTiger is stored as a table in the \c "WiredTiger.wt" file.
-The table's key is a \c uri string, and the value is its complete
-configuration string.  The configuration itself is a list of key/value
-pairs in string form.  The configuration's keys are dependent on the
-type of \c uri.  Thus, a metadata entry with a \c uri key beginning with
-\c "table:" will be a configuration string
-having configuration entries like \c key_format and \c value_format to
-describe the data encoding for the uri.  A metadata key beginning with
+Metadata in WiredTiger captures important information about the users database.
+Metadata includes tracking essential information such as the files and tables
+present in the database, their associated configurations, as well as the latest checkpoints of
+each those files.
+
+The main metadata in WiredTiger is stored in the \c WiredTiger.wt file. The
+\c WiredTiger.wt file being a table itself.
+
+@section metadata_table Metadata Table
+
+The keys in the \c WiredTiger.wt table are \c uri strings. The \c uri keys can be strings
+representing:
+- WiredTiger Tables and Files: \c "table:" & \c "file:"
+- Indexes: \c "index:"
+- Colgroups: \c "colgroup:"
+- System values: \c "system". Representing special system values.
+
+The value corresponding to a given \c uri key in the metadata table is its
+associated configuration string.  The configuration value itself is a list of key/value
+pairs in string form.  The key/value pairs in the configuration strings are dependent on the
+type of \c uri.  Thus, a metadata entry with a \c uri key beginning with \c "table:" will be
+a configuration string having entries such as \c key_format and \c value_format to
+describe the data encoding for the uri. A metadata key beginning with
 \c "file:" will have a different set of configuration entries associated
 with it.
 
-A caller of WiredTiger can use WT_SESSION::open_cursor with a \c uri equal to
-\c "meta:" to read the metadata.  Using this interface, metadata can only
-be queried, not changed.  Changes to the metadata are affected by API calls
-such as WT_SESSION::create, WT_SESSION::drop, WT_SESSION::rename.
+@section metadata_checkpoint Storing Checkpoint Metadata
+
+Metadata plays an important role in the WiredTiger checkpointing process. Metadata is used to store
+the latest checkpoint information for each btree in the system. The checkpoint metadata providing a known
+point in time in which WiredTiger can recover from (e.g after a crash or unexpected shutdown).
```

Comment:
These three sentences feel reversed to me (and the 3rd is not a complete sentence). Maybe something like:
(remove the first sentence, modify the other two) 
The latest checkpoint information for each btree in the system is stored in its metadata entry. The checkpoint metadata provides a known point in time from which WiredTiger can access the data after a restart.

Reply:
Thanks for the feedback on the readability of those sentences. I've reworked per your suggestion: 1d7b024eee72636eb71d9e0359041f67ee3304c3